upcloud/service/storage.go

```diff
@@ -428,3 +429,27 @@ func (s *Service) WaitForStorageImportCompletion(r *request.WaitForStorageImport
 		}
 	}
 }
+
+// ResizeStorage resizes the last partition of a storage and the ext3/ext4/XFS/NTFS filesystem
+// on that partition if the partition does not extend to the end of the storage yet.
+//
+// Before the resize is attempted, a backup is taken from the storage. If the resize
+// succeeds, backup details are returned. It is advisable to keep the backup until
+// you have ensured that everything works after the resize.
+//
+// If the resize fails, backup is used to restore the storage to the state where it
+// was before the resize. After that the backup is deleted automatically.
+func (s *Service) ResizeStorage(r *request.ResizeStorageRequest) (*upcloud.StorageResizeBackup, error) {
```

Comment:
Perhaps we should name it `ResizeStoragePartition`? `ResizeStorage` alone suggests that this function takes care of all required storage modifications, while in reality user has to do a modify storage request (with size change) first.

Reply:
I agree that the name is not most descriptive but I think that on the other hand it follows the rest of the naming convention because that URL is `/storage/<UUID>/resize` . I think that technically name should be something like `ResizeStoragePartitionAndFilesystem` but then it wouldn't map nicely to our API docs. For me it would make more sense to change the API URL first although I know that it might be easier said than done :)

Well that's my two cents if you and @Darep think that it is better to rename I can do it.

Reply:
It is tricky… For anybody just browsing through the code or docs, `ResizeStorageFilesystem` as the name (or `ResizeStoragePartition` or both) would give the best hint as to what that method is all about. Having it as `ResizeStorage` might make some users call it without calling ModifyStorage with `size` first since they might think ResizeStorage does both. Though I feel like you could make the same mistake even if it was `ResizeStorageFilesystem` so then, in the end, the name might not matter :joy:

Reply:
What if we add new (e.g. `ModifyStorageSize`) function that would do modify and resize?

Reply:
Hmm, no strong feelings, but maybe we shouldn't go there as it would be one more thing to maintain, and things are more flexible with the separate methods 😊

Reply:
I wouldn't mind adding that, but even then I would keep the `ResizeStorage(PartitionAndFilesystem)` function exposed to the public. It could be beneficial in some cases, like in TF provider.

As far as the name goes, I get the argument about the mapping to API, but tbh the api endpoint is kind of misleading here, and the point of SDK is to make it easier to use API. We can keep this name of course, but then I would add some additional explanation to the doc comment, to make it absolutely explicit that this function is for resizing the partition and filesystem after the storage itself has been resized.

Reply:
Thanks for comments! Renamed this to ResizeStorageFilesystem as it is more descriptive name.